README.md

```diff
@@ -1,5 +1,145 @@
+a by the server defined version identifier. This allows firstly to easily
+identify if a client needs to fetch a newer version. Secondly, it allows a web
+application to decide that a manifest cached in the client is "good enough" to
+avoid fetching a newer version for performance reasons.
+
+
+### Fetching Protocol
+Clients always set the `Sec-Origin-Manifest` header to indicate the support for
+the mechanism. Servers can then opt-in to use the mechanism by sending back a
+`Sec-Origin-Manifest` header with the current manifest version. If servers
+should decide to not send the header, clients try to retreive a manifest from
+their cache and not use Origin Manifest otherwise.
+
+#### Opt-in
+In case no manifest is cached the client indicates support for the feature by
+setting the value 1.
+```
+Client                                                      Server
+  |    --- request with 'Sec-Origin-Manifest: 1' --------->   |
+  |    <-- response with 'Sec-Origin-Manifest: "v0"' ------   |
+  | defer processing response
+  |    --- request to '<origin>/originmanifest/v0.json' -->   |
+  |    <-- response with version0.json --------------------   |
+  | continue processing first response
+```
+
+#### Updating and Confirming
+In case a manifest is cached the client informs the server about the currently
+cached version. If the version is accepted by the server the response simply
+confirms the header. No fetch is needed. Otherwise the new version is fetched.
+```
+Client                                                      Server
+  |    --- request with 'Sec-Origin-Manifest: "v0"' ------>   |
+  |    <-- response with 'Sec-Origin-Manifest: "v1"' ------   |
+  | defer processing response
+  |    --- request to '<origin>/originmanifest/v1.json' -->   |
+  |    <-- response with v1.json --------------------------   |
+  | continue processing first response
+```
+
+#### Opt-out
+Servers can decide not no longer use Origin Manifest. If so they can set the
+`Sec-Origin-Manifest` header to 0 in the response. Clients then behave like no
+manifest was ever set and remove the currently cached manifest from cache (if
+any).
+
+## Discussion
+**Why is the current version sent?**
+
+The currently cached version is set in the request header to allow HTTP2 Push.
+In particular, the server can decide to push the manifest since it knows whether
+it will be eventually fetched or not.
+
+
+**Why does it defer the response?**
+
+Certain configuration options, e.g. CSP, require to be loaded before the actual
+content is processed, e.g. a HTML document is rendered. This makes it also
+fundamentally different from Application Manifest (TODO: ref here).
+
+
+**Why not just using HTTP caching and ETag?**
+
+We need to process Origin Manifests differently from other data fetched over
+HTTP. In particular, it allows us to manage the different versions and to ensure
+that there exists at most only exactly one manifest per origin.
```

Comment:
This doesn't really answer the question. :) I'd suggest diving in a little deeper: both etags and caching require the server to start a push that the client will cancel. This has a performance cost that it's not clear we should ask clients to pay.